chainbase/src/main/java/org/tron/core/db2/core/SnapshotImpl.java

```diff
@@ -48,21 +54,36 @@ public byte[] get(byte[] key) {
   }
 
   private byte[] get(Snapshot head, byte[] key) {
+    Histogram.Timer requestTimer = Metrics.histogramStartTimer(
+            MetricKeys.Histogram.SNAPSHOT_SERVICE_LATENCY, root.getDbName(), MetricLabels.Histogram.SNAPSHOT_GET);
     Snapshot snapshot = head;
     Value value;
     if (isOptimized) {
       value = db.get(Key.of(key));
+      if(Objects.nonNull(value)){
+        Metrics.histogramObserve(requestTimer);
+      }
```

Comment:
if null, not record?

Reply:
if null,the result will be recorded when the query from snapshotRoot completed.

Reply:
> if null, not record?

if null,the result will be recorded when the query from snapshotRoot completed.